tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,167 @@
+__all__ = ["FastTextModel"]
+
+
+import logging
+import tempfile
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+from pandas.api.types import is_numeric_dtype
+
+from ...features.feature_metadata import FeatureMetadata
+
+from autogluon.core.utils.loaders import load_pkl
+from autogluon.core.utils.savers import save_pkl
+
+from ...constants import BINARY, MULTICLASS
+from ..abstract.abstract_model import AbstractModel
+from .hyperparameters.parameters import get_param_baseline
+
+logging.basicConfig(
+    format="%(asctime)s: [%(funcName)s] %(message)s", level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+
+def try_import_fasttext():
+    try:
+        import fasttext
+
+        _ = fasttext.__file__
+    except Exception:
+        raise ImportError('Import fasttext failed. Please run "pip install fasttext"')
+
+
+class FastTextModel(AbstractModel):
+    model_bin_file_name = "fasttext.bin"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        if self.problem_type not in (BINARY, MULTICLASS):
+            raise ValueError(
+                "FastText model only supports binary or multiclass classification"
+            )
+
+    def _set_default_params(self):
+        default_params = get_param_baseline()
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _fit(self, X_train, y_train, **kwargs):
+        try_import_fasttext()
+        import fasttext
+
+        if self.feature_metadata is None:
+            self.feature_metadata = self._build_feature_metadata(X_train)
+        self.text_features = self.feature_metadata.get_features(valid_raw_types=['object', 'category'])
```

Comment:
I'll be working on a PR to make this easier soon, so it is reflected in self.features

Reply:
PR is up: #756 , this should simplify this process.

After #756 is merged, you can simply add:

```
    def _get_default_auxiliary_params(self) -> dict:
        default_auxiliary_params = super()._get_default_auxiliary_params()
        extra_auxiliary_params = dict(
            get_feature_kwargs=dict(
                valid_raw_types=[R_OBJECT],
            )
        )
        default_auxiliary_params.update(extra_auxiliary_params)
        return default_auxiliary_params
```

And get rid of self.text_features. self.features will have everything you need and the input to _fit will already have the DataFrame filtered to only the text features (aka the object features).